### ISSUE TITLE
Amend GetTimestampAtHeight to accept ConnectionI interface

### ISSUE THREAD

**Issue by DimitrisJim** [cosmos/ibc-go#4021]
## Summary

The `connectionKeeper`'s `GetTimestampAtHeight` method currently expects a `ConnectionEnd` type. This is in contrast to the rest of the methods which are defined based on the `ConnectionI` interface defined in `exported`.

This results in a requirement to mix calls (see #3840) to access connection keeper methods since `channelKeeper.GetConnection` returns the interface when `GetTimestampAtHeight` requires the type (which is accessed via `connectionKeeper.GetConnection`).

## Proposal

Change `GetTimestampAtHeight`'s sig to accept the interface instead of the type.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by DimitrisJim** [comment]
note: unsure if there's history behind this? Also, pretty sure this is API-breaking, though I wouldn't expect anyone to actually be using it.

**Comment by crodriguezvega** [comment]
Good catch, @DimitrisJim! it could be that maybe this is even not API breaking? [The `ConnectionEnd` fully implements the `ConnectionI` interface](https://github.com/cosmos/ibc-go/blob/1fda14dd18b24e08a750edb2e3eae8d1465fe2f9/modules/core/03-connection/types/connection.go#L12-L48), so maybe it's possible to just replace the argument type without any harm.

**Comment by DimitrisJim** [comment]
due to it being a public symbol it will necessarily be API-breaking I'd assume. People do wild shit with code. I do agree that yeah, call sites won't be affected due to the ConnectionEnd implementing that interface, on our end this just changes signatures.
